hamocc/mo_sedmnt.F90

```diff
@@ -97,16 +105,185 @@ MODULE mo_sedmnt
+        write(io_stdo_bgc,'(5F9.3)') dzs
+        write(io_stdo_bgc,*)  ' '
+      endif
+
+      seddzi(1) = 500.
+      do k = 1, ks
+        seddzi(k+1) = 1. / dzs(k+1)
+        seddw(k) = 0.5 * (dzs(k) + dzs(k+1))
+      enddo
+
+#ifndef sedbypass
+      ! 2d and 3d fields are not allocated in case of sedbypass
+      ! so only initialize them if we are using the sediment
+      CALL ini_sedmnt_fields(kpie,kpje,kpke,omask,sed_por)
+#endif     
+ END SUBROUTINE ini_sedmnt
+
+      !========================================================================
+ SUBROUTINE ini_sedmnt_fields(kpie,kpje,kpke,omask,sed_por)
+      use mo_control_bgc, only: l_3Dvarsedpor
+
+      implicit none
+
+      integer, intent(in) :: kpie,kpje,kpke
+      real,    intent(in) :: omask(kpie,kpje)
+      real,    intent(in) :: sed_por(kpie,kpje,ks)
+
+      ! local
+      integer :: i,j,k
+
+      ! this initialization can be done later via reading a porosity map
+      if (l_3Dvarsedpor)then
+       ! lon-lat variable sediment porosity from input file
+       do k=1,ks
+       do j=1,kpje
+       do i=1,kpie
+        if(omask(i,j).gt. 0.5)then
+          porwat(i,j,k) = sed_por(i,j,k)
+        endif
+       enddo
+       enddo
+       enddo
+      else
+        porwat(:,:,1) = 0.85
+        porwat(:,:,2) = 0.83
+        porwat(:,:,3) = 0.8
+        porwat(:,:,4) = 0.79
+        porwat(:,:,5) = 0.77
+        porwat(:,:,6) = 0.75
+        porwat(:,:,7) = 0.73
+        porwat(:,:,8) = 0.7
+        porwat(:,:,9) = 0.68
+        porwat(:,:,10) = 0.66
+        porwat(:,:,11) = 0.64
+        porwat(:,:,12) = 0.62
+      endif
+
+      if (mnproc == 1) then
+        write(io_stdo_bgc,*)  'Pore water in sediment initialized'
+      endif
+
+      do k = 1, ks
+      do j = 1, kpje
+      do i = 1, kpie
+        porsol(i,j,k) = 1. - porwat(i,j,k)
+        if(k >= 2) porwah(i,j,k) = 0.5 * (porwat(i,j,k) + porwat(i,j,k-1))
+        if(k == 1) porwah(i,j,k) = 0.5 * (1. + porwat(i,j,1))
+      enddo
+      enddo
+      enddo
+
+      ! determine total solid sediment volume
+      solfu = 0.
+      do i = 1, kpie
+      do j = 1, kpje
+      do k = 1, ks
+        solfu(i,j) = solfu(i,j) + seddw(k) * porsol(i,j,k)
+      enddo
+      enddo
+      enddo
+
+      ! Initialize porosity-dependent diffusion coefficients of sediment
+      zcoefsu(:,:,0) = 0.0
+      do k = 1,ks
+      do j = 1, kpje
+      do i = 1, kpie
+        ! sediment diffusion coefficient * 1/dz * fraction of pore water at half depths
+        zcoefsu(i,j,k  ) = -sedict * seddzi(k) * porwah(i,j,k)
+        zcoeflo(i,j,k-1) = -sedict * seddzi(k) * porwah(i,j,k)    ! why the same ?
+      enddo
+      enddo
+      enddo
+      zcoeflo(:,:,ks) = 0.0                    ! diffusion coefficient for bottom sediment layer
       
+      if (mnproc == 1) then
+        write(io_stdo_bgc,*)  'Pore water diffusion coefficients in sediment initialized'
+      endif
+
+ END SUBROUTINE ini_sedmnt_fields
```

Comment:
Wouldn't a more specific name be better, e.g. `ini_sedmnt_por` or something like this? `ini_sedmnt_fields` makes me wonder (if I were new to the code) what the difference between this routine and `alloc_mem_sedmnt` is.

Reply:
Ok, I agree the difference between 'ini' and 'alloc' shoud be clear (allthough the former could include the latter). I just thought that the routine is dealing with initializing sediment porosity (and related/derived quantities) so '_por'  or similar instead of '_fields' would be more specific. But not so important after all.

Reply:
Ok, I renamed it to `_por` and provide a bit more information what will be initialized.

Reply:
Hi @TomasTorsvik and @JorgSchwinger , I now checked the feature branch against the `master` - in case of `sedbypass=true`, the latest `master` didn't went through - so there the bug caused issues (sorry again) - so don't use sha: 88d6208d053733a87dec9546be6a2287f8626941 in cases for `sedbypass=true` - I re-ran the `master` with sha: 3e4eb0f4543175dce3b92a070ea454cdfec1f645 (the commit before introducing the variable porosity) for testing purposes and major nutrients and alkalinity are identical in both cases (`sebypass=true` and `sedbypass=false`). So I'll merge the branch.